react-catalog/my-app/src/Catalog.tsx

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from 'react';
+import { Product, readCatalog, toDollars } from './lib';
+import { Link } from 'react-router-dom';
+
+export function Catalog() {
+  const [items, setItems] = useState<Product[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+  useEffect(() => {
+    async function loadItems(): Promise<void> {
+      try {
+        const values = await readCatalog();
+        setItems(values);
+        console.log(values);
+      } catch (error) {
+        setError(error);
+        console.error('Error: ', error);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    loadItems();
+  }, []);
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
+  if (error) {
+    return (
+      <div>
+        Error! {error instanceof Error ? error.message : 'Unknown Error'}
+      </div>
+    );
+  }
+  function catalogCards(item: Product) {
```

Comment:
I think this should be moved outside of the Catalog function (you can just put it below in this same file) and become its own component. That would make it more reusable and easier to work with.